File: 112-path-sum/path-sum.cpp
class Solution {
public:
    bool solve(TreeNode* root, int tg){
        if(!root) return false;

        if(root->val == tg && !root->left && !root->right) return true;

        bool l = solve(root->left, tg - root->val);
        bool r = solve(root->right, tg - root->val);

        return l || r;
    } 
    bool hasPathSum(TreeNode* root, int targetSum) {
        return solve(root, targetSum);
    }
};

File: 112-path-sum/path-sum.java
class Solution {
    public boolean hasPathSum(TreeNode node, int sum) {
        if (node == null) {
            return false;
        }

       
        if (node.left == null && node.right == null && sum == node.val){
            return true;
        }

        // Recur down the left and right subtrees
        int remSum = sum - node.val;
        
        return hasPathSum(node.left, remSum) || hasPathSum(node.right, remSum);
    }
}
